### PR TITLE
gargoyle: 2018.10.06 -> 2019.01.01

### PR DESCRIPTION
**Pull request by emacsomancer**
- latest released version

###### Motivation for this change
update to latest release

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

##### Notes
- there are more commits since the last release, including improving some things for macOS, so building from one of the later post-release commits could have advantages. I don't have a mac to test on one way or other though.

### CODE REVIEW

**Comment by drewrisinger** [comment]
Build fails locally due to incorrect hash for GitHub source.

@GrahamcOfBorg build gargoyle

**Comment by emacsomancer** [comment]
> Build fails locally due to incorrect hash for GitHub source.
> 
> @GrahamcOfBorg build gargoyle

I think I must not be deriving it properly then. What's the best method of doing this?

**Comment by drewrisinger** [comment]
I think the proper way is something like (from command-line)

```bash
nix-prefetch-git
```
I usually just go the lazy way and copy the SHA256 that Nix puts on the command line though.

**Comment by drewrisinger** [comment]
You should always try to build it locally via ``nix-build -A gargoyle`` before pushing a PR. saves everyone time. And if you can't for some reason, then mark your PR as draft and ask OfBorg (i.e. GrahamcOfBorg like I did above) to build your package, though it's nice to limit your builds due to OfBorg being a shared resource.

**Comment by emacsomancer** [comment]
> You should always try to build it locally via `nix-build -A gargoyle` before pushing a PR. saves everyone time. And if you can't for some reason, then mark your PR as draft and ask OfBorg (i.e. GrahamcOfBorg like I did above) to build your package, though it's nice to limit your builds due to OfBorg being a shared resource.

For some reason I must have altered the sha256 after building, but before pushing, because I've been running 2019.01.01 on my system locally for a while now.

**Comment by drewrisinger** [comment]
Strange. I can try to review sometime if you update the hash.

**Comment by emacsomancer** [comment]
> Strange. I can try to review sometime if you update the hash.

I already did (before replying), and force pushed the branch.

**Comment by drewrisinger** [comment]
OK. GitHub email doesn't notify on force push.

**Review by drewrisinger — APPROVED** [review]
Passes ``nixpkgs-review pr 85576``

Not tested locally.
```
https://github.com/NixOS/nixpkgs/pull/85576
1 package built:
gargoyle
```

File: pkgs/games/gargoyle/default.nix [hunk 1]
```diff
@@ -20,13 +20,13 @@ let
 in
 
 stdenv.mkDerivation {
-  name = "gargoyle-2018-10-06";
+  name = "gargoyle-2019-01-01";
```
Comment (danieldk) [inline pkgs/games/gargoyle/default.nix:23]:
```suggestion
  pname = "gargoyle";
  version = "2019.1.1";
```

I checked the revision and it actually seems to respond to the tag `2019.1.1`, so it seems to be better to use that as the release. Besides that the release date was Nov 18, 2019  ;).

This change is also a good excuse to switch to the newer `pname`/`version` convention.

File: pkgs/games/gargoyle/default.nix [hunk 2]
```diff
@@ -20,13 +20,13 @@ let
 in
 
 stdenv.mkDerivation {
```
Comment (danieldk) [inline pkgs/games/gargoyle/default.nix:22]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/games/gargoyle/default.nix [hunk 3]
```diff
@@ -20,13 +20,13 @@ let
...
-  name = "gargoyle-2018-10-06";
+  name = "gargoyle-2019-01-01";
 
   src = fetchFromGitHub {
     owner = "garglk";
     repo = "garglk";
-    rev = "d03391563fa75942fbf8f8deeeacf3a8be9fc3b0";
-    sha256 = "0icwgc25gp7krq6zf66hljydc6vps6bb4knywnrfgnfcmcalqqx9";
+    rev = "28b999f83542deee60f2447e4b6109b6db358f57";
```
Comment (danieldk) [inline pkgs/games/gargoyle/default.nix:28]:
```suggestion
    rev = version;
```

**Review by danieldk — CHANGES_REQUESTED** [review]

File: pkgs/games/gargoyle/default.nix (hunk 3 above)
Comment (emacsomancer) [reply pkgs/games/gargoyle/default.nix:28]: I think I must have implemented some suggestion wrong, but I get:

    error: fetchurl requires a hash for fixed-output derivation: https://github.com/garglk/garglk/archive/2019.1.1.tar.gz

Comment (danieldk) [reply pkgs/games/gargoyle/default.nix:28]: You removed the `sha256` attribute.

Comment (danieldk) [reply pkgs/games/gargoyle/default.nix:28]: @emacsomancer let me know if you need any help. I can also update the PR to make these changes if you prefer.

Comment (emacsomancer) [reply pkgs/games/gargoyle/default.nix]: thanks. I've pushed the changes now.

File: pkgs/games/gargoyle/default.nix [hunk 4]
```diff
@@ -19,14 +19,15 @@ let
 
 in
 
-stdenv.mkDerivation {
-  name = "gargoyle-2018-10-06";
+stdenv.mkDerivation rec {
+  name = "gargoyle";
```
Comment (danieldk) [inline pkgs/games/gargoyle/default.nix:23]:
```suggestion
  pname = "gargoyle";
```

Note that this should be `pname`, not `name`. `name` is defined in terms of `pname` and `version`.

File: pkgs/games/gargoyle/default.nix [hunk 5]
```diff
@@ -19,14 +19,15 @@ let
...
+  version = "2019.1.1";
 
   src = fetchFromGitHub {
     owner = "garglk";
     repo = "garglk";
-    rev = "d03391563fa75942fbf8f8deeeacf3a8be9fc3b0";
     sha256 = "0icwgc25gp7krq6zf66hljydc6vps6bb4knywnrfgnfcmcalqqx9";
+    rev = version;
```
Comment (danieldk) [inline pkgs/games/gargoyle/default.nix:30]: Could you move the `rev` attribute between the `repo` and `sha256` attributes? Doesn't change the evaluation of the derivation, but this is the more canonical ordering in `nixpkgs`.

**Review by danieldk — CHANGES_REQUESTED** [review]
Thanks for the changes!

Looks almost good to go. It looks like you accidentally used `name` in place of `pname`. Also one small comment about the ordering in `src`.

**Comment by danieldk** [comment]
Somewhere along the way, you switched the hash (`sha256`) to that of the previous version. I am getting

`0w54avmbp4i4zps2rb4acmpa641s6wvwbrln4vbdhcz97fx48nzz`

as the hash.

**Review by danieldk — APPROVED** [review]
LGTM, thanks!

Result of `nixpkgs-review pr 85576` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- gargoyle
</details>

Tested running gargoyle.

**Pull request merged**